rust/processor/src/models/token_models/ans_lookup.rs

```diff
@@ -74,107 +77,225 @@ impl OptionalString {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordKeyV1 {
+    domain_name: String,
+    subdomain_name: OptionalString,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct NameRecordV1 {
+    #[serde(deserialize_with = "deserialize_from_string")]
+    expiration_time_sec: BigDecimal,
+    #[serde(deserialize_with = "deserialize_from_string")]
+    property_version: BigDecimal,
+    target_address: OptionalString,
+}
+
 impl CurrentAnsLookup {
     pub fn from_transaction(
```

Comment:
Can you add some comments on what the overall flow is? It's a bit difficult understanding how these different handles play with each other.

Reply:
ping